converter/codec.go

```diff
@@ -364,7 +364,7 @@ type remoteDataConverter struct {
 
 // NewRemoteDataConverter wraps the given parent DataConverter and performs
 // encoding/decoding on the payload via the remote endpoint.
-func NewRemoteDataConverter(parent DataConverter, options RemoteDataConverterOptions) DataConverter {
+func NewRemoteDataConverter(parent DataConverter, options RemoteDataConverterOptions) *remoteDataConverter {
```

Comment:
Generally you don't return an unexported type from an exported function

Reply:
OK, I'm not a Go expert -- I tentatively thought that it was reasonable in this case as a way of communicating to the caller that they can choose which interface they want to interpret the return value as implementing. But also, I was trying to make as minimal changes as possible in order to solicit initial feedback on the overall approach. I'll export the type if that's the right thing to do.

Reply:
> I tentatively thought that it was reasonable in this case as a way of communicating to the caller that they can choose which interface they want to interpret the return value as implementing

The type does not need to be unexported for that to be the case. Unexported types are harder to work with and I believe the methods it implements won't show up on the docs so it can be hard to tell what it implements.

Reply:
Yep. I've exported it, and also changed the docstring to say "RemoteDataConverter is a DataConverter and PayloadCodec ..." 029d5aa97357d039d98c07ea6348debcf8aa260b

Reply:
What I haven't done yet is look into refactoring it cleanly as a standalone `PayloadCodec` used by a `DataConverter`, because I wanted to get an OK for this approach without making too many code changes. Not sure if you feel that's necessary or a future nice-to-have.

Reply:
@cretz if I do that do you think you would use it in the next version of the CLI? Do you envisage using a gRPC interceptor for payload encoding/decoding alone, and not for ser/de? (I'm thinking yes, but wanted to check).